src/hls_lpdaac_reconciliation/generate_report/index.py

```diff
@@ -191,59 +148,49 @@ def generate_report(
         prefixes will be included ("S30", "L30", "S30_VI", "L30_VI").
     """
-    athena = AthenaQueryClient(catalog_name=catalog_name, database_name=database_name)
-
-    product_prefixes_regex = f"({'|'.join(product_prefixes)})"
-    sql = f"""
-    WITH inventory AS (
-        SELECT *
-        FROM {table_name}
-        WHERE dt = (SELECT max(dt) FROM {table_name})
-            AND last_modified_date BETWEEN
-                TIMESTAMP '{start_date:%Y-%m-%d}'
-                AND TIMESTAMP '{end_date:%Y-%m-%d}'
-    ),
-    parsed AS (
-        SELECT regexp_extract(key, '[^/]+$') as filename,
-            size,
-            date_format(last_modified_date, '%Y-%m-%dT%H:%i:%sZ') AS last_modified,
-            'NA' as checksum
-        FROM inventory
-        WHERE REGEXP_LIKE(
-                key,
-                '^{product_prefixes_regex}/data/.*(tif|jpg|xml|stac.json)$'
-            )
-    )
+    print(f"Generating report for {start_date} to {end_date}")
+
+    product_prefixes = product_prefixes or _DEFAULT_PRODUCT_PREFIXES
+    key_pattern = rf"^({'|'.join(product_prefixes)})/data/.*(tif|jpg|xml|stac\.json)$"
+    sql = rf"""
     SELECT
-        REGEXP_REPLACE(
-            filename,
-            '([aA-zZ0-9]+)\\.([aA-zZ0-9]+).*',
-            '$1$2'
-        ) AS short_name,
-        REGEXP_EXTRACT(filename, 'v([0-9]\\.[0-9])', 1) AS version,
-        filename,
+        -- short_name is the concatenation of "HLS" and the first component of the key.
+        -- For example, "HLSL30_VI" for key "L30_VI/data/2025067/HLS-VI.L30.T...".
+        regexp_replace(key, '^([^/]+).*', 'HLS$1') AS short_name,
+        regexp_extract(key, 'v([0-9]+(?:\.[0-9]+)*)', 1) AS version,
+        regexp_extract(key, '[^/]+$') AS filename,
         size,
-        last_modified,
-        checksum
-    FROM parsed
+        -- Use format_datetime, which allows us to include 3 decimal places for seconds
+        -- value (based upon Java's formatting capabilities), whereas date_format
+        -- supports only 6 decimal places (based upon Python's formatting).  This is to
+        -- guarantee we output timestamps formatted exactly as previously done, so that
+        -- we don't potentially cause problems on the LP DAAC side that parses it.
+        -- See https://trino.io/docs/current/functions/datetime.html#java-date-functions
+        format_datetime(last_modified_date, 'yyyy-MM-dd''T''HH:mm:ss.SSS''Z''') AS last_modified,
+        'NA' as checksum
+    FROM {table}
+    WHERE dt = (SELECT max(dt) FROM {table})
+        AND last_modified_date BETWEEN
+            TIMESTAMP '{start_date:%Y-%m-%d}'
+            AND TIMESTAMP '{end_date:%Y-%m-%d}'
+        AND regexp_like(key, '{key_pattern}')
+    ORDER BY last_modified_date
     """
 
-    result = athena.execute_query(
+    # athena = AthenaQueryClient(catalog_name=catalog_name, database_name=database_name)
+
+    query_id = execute_query(
```

Comment:
nit - commented out old line

```suggestion
    query_id = execute_query(
```